lib/order.rb

```diff
@@ -0,0 +1,66 @@
+require 'csv'
+require_relative 'customer'
+
+class Order
+  attr_reader :id, :products, :customer, :fulfillment_status
+
+  def initialize(id, product_hash, customer, fulfillment_status = :pending)
+    @id = id
+    @products = product_hash
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+
+    valid_statuses = %i[pending paid processing shipped complete]
+
+    unless valid_statuses.include?(@fulfillment_status)
+      raise ArgumentError, 'Not a valid order'
+    end
+  end
+
+  def total
+    sum = @products.values.sum
+    return (sum*1.075).round(2)
+  end
+
+  def add_product(name, price)
+    if @products.has_key?(name)
+      raise ArgumentError, 'This product already exists in order!'
+    end
+    return @products.merge!(name => price)
```

Comment:
Nice use of `.merge!`! Hadn't seen anyone use this elsewhere and I think this is a great time to use it.